module-2/backend/week_three.md

```diff
@@ -10,14 +10,35 @@ Note: When you're done, submit a PR with a reflection in the comments about how
 ### Questions
 
 1. What is the entry at the command line to create a new rails app?
+  rails new AppName -T --database=postgresql --skip-turbolinks --skip-spring
 2. What do Models generally inherit from in rails?
+  ActiveRecord::Base
 3. What do Controllers generally inherit from in a rails project?
+  ApplicationController
 4. How would I create a route if I wanted to see a specific horse in my routes fitle assuming I'm sticking to standard conventions and that I didn't want other CRUD functionality?
+  resources :horses, only: [:show]
 5. What rake task is useful when looking at routes, and what information does it give you?
+  rake routes: path, uri, controller#action
 6. What is an example of a route helper? When would you use them?
+  resources :horses, as: 'awesome_horses'
+  if you had a normal resource of horses, that weren't awesome and you want an extra degree of separation between the two
 7. What's the difference between what `_url` and `_path` return when combined with a routes prefix?
+  path is relative, url is absolute
 8. What are strong params and why are the necessary?
+  it's how rails makes you import input from forms, helps with security, prevents malicious code from being input
 9. What role does `form_for` play in helping us create our forms?
+  it creates the form, and the path to which the data is sent
 10. How does `form_for` know where to submit the user's input?
+  some black magic (for edit v new), but you also define it in the first line
```

Comment:
Ha! Edit vs. new is determined based on whether the argument has been saved to the database or not. So, if `@horse` in the example below is `Horse.new`, Rails will check and see that it has not been saved to the database, so will create a form that will send a POST request to `/horses`. On the other hand, if `@horse` is `Horse.find(1)`, it will check and find that the horse *has* been saved to the DB, and will create a form that submits a PATCH request to `horses/1`.